integreat_cms/cms/views/regions/region_actions.py

```diff
@@ -37,6 +38,51 @@ def delete_region(request, *args, **kwargs):
     """
 
     region = get_object_or_404(Region, slug=kwargs.get("slug"))
+
+    # Check whether region can be safely deleted
+    mirrored_pages = Page.objects.filter(
+        region=region, mirroring_pages__isnull=False
+    ).distinct()
+    if len(mirrored_pages) > 0:
```

Comment:
Since a lot of database operations are performed before the Protection error occurs, I would suggest to check in advance whether the deletion can be executed. Otherwise the region would still exist but be in an inconsistent state (e.g. the complete page hierarchy would be lost, the media library etc...)